Update/Add new type classes and CJK Fonts

jordanjones243 opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #269, #272
https://dev.azure.com/itsals/E_Retain_Content/_workitems/edit/1524213
## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)


## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Add emphasized body text variations with heavier weights across all themes, update CSS variable generation for per-size font weight, adjust heading weights for Alaska and Hawaiian themes, remove redundant letter-spacing in Hawaiian accent config, bump design-tokens dependency and docs version.

New Features:
- Introduce default-emphasized, lg-emphasized, sm-emphasized, xs-emphasized, and 2xs-emphasized body text classes across all themes

Enhancements:
- Refactor body mixin to generate CSS variables for font weight per size instead of a single weight variable
- Adjust heading weight values in Alaska and Hawaiian theme configurations
- Remove letter-spacing settings from Hawaiian accent type configuration

Build:
- Upgrade @aurodesignsystem/design-tokens dependency to ^8.6.0

Documentation:
- Bump documentation site version from 10.1.3 to 10.1.4

jordanjones243 committed:
perf: update values in header classes #269

jordanjones243 committed:
perf: remove letter-spacing from hawaiian accents #269

BREAKING CHANGE: The letter-spacing rule is now omitted from the Hawaiian theme `accent` classes.

jordanjones243 committed:
feat: add new body-emphasized classes #269

chrisfalaska commented:
I did some diffs based on the current Prod version.

Is this what is expected @jordanjones243 & @forrest-for-real ?

- Alaska Classic - https://www.diffchecker.com/B3YjBg33/

- Alaska - https://www.diffchecker.com/pfPEbAQF/

- Auro-1: https://www.diffchecker.com/aRgdqNHr/

- Hawaiian - https://www.diffchecker.com/ID57MWo4/

chrisfalaska commented:
> I did some diffs based on the current Prod version.
> 
> Is this what is expected @jordanjones243 & @forrest-for-real ?
> 
> * Alaska Classic - https://www.diffchecker.com/B3YjBg33/
> * Alaska - https://www.diffchecker.com/pfPEbAQF/
> * Auro-1: https://www.diffchecker.com/aRgdqNHr/
> * Hawaiian - https://www.diffchecker.com/ID57MWo4/

After reviewing the diff links with @forrest-for-real , we realized there is a font dependency and created https://github.com/AlaskaAirlines/WebCoreStyleSheets/issues/272

jordanjones243 committed:
perf: ensure figma and type structure align after 8.18 ADT release #269

jordanjones243 committed:
build: update design token package reference in postinstall

forrest-for-real reviewed (dismissed):
I reviewed changes live with @jordanjones243 and it looks good from my end.

chrisfalaska commented:
I ran a diff against `master`:

[alaska.global.css](https://www.diffchecker.com/yURIIFL3/)

Are these indicative of the changes you are expecting?

cc: @forrest-for-real

forrest-for-real commented:
> I ran a diff against `master`:
> 
> [alaska.global.css](https://www.diffchecker.com/yURIIFL3/)
> 
> Are these indicative of the changes you are expecting?
> 
> cc: @forrest-for-real

Is this for the latest build that @jordanjones243 did today @chrisfalaska? Your diff is showing the body-emphasized font weight as 700, and it should be 500 after the updates made today.

jordanjones243 committed:
feat: add 500 med font-face for CJK fonts #272

jordanjones243 commented:
@forrest-for-real @chrisfalaska 
Diff check for Alaska, Hawaiian & Atmos themes with latest commit for CJK fonts:

Alaska: https://www.diffchecker.com/oNIbjSLT/
Hawaiian: https://www.diffchecker.com/vdKO43ko/
Atmos/Auro-1: https://www.diffchecker.com/aF4kXudM/

chrisfalaska commented on packageScripts/postinstall.mjs:
```diff
@@ -23,7 +23,7 @@ chalk.hex('#f26135')(`
                of `) + chalk.hex('#ffd200').bold(`WC Stylesheets v${pjson.version}.`) + chalk.hex('#f26135')(`
 
-            Requires Design Tokens ${pjson.peerDependencies["@aurodesignsystem/design-tokens"]}
+            Requires Design Tokens ${pjson.dependencies["@aurodesignsystem/design-tokens"]}
```
Just want to confirm this was intentional? Not sure what this change represents specifically.

jordanjones243 replied:
This was intentional, yes. When running a `build`, WCSS tells you what design tokens it depends on. At some point, the dependency got moved from peer to direct and this path never got updated.

chrisfalaska commented on src/type/mixins/_fluid-type.scss:
```diff
@@ -23,31 +27,35 @@
+          auro_map-deep-get($fallback-config, 'sizes', $size, 'font-size', 'max')
+        )});
+
+        // Conditionally exclude letter-spacing for Hawaiian theme
```
Since this repeats across themes, might be worth splitting out into a mixin or func

chrisfalaska commented on src/type/partials/_body.scss:
```diff
@@ -3,16 +3,16 @@
 @import '../../utilityVariables/cssNamespace';
 
-$body-sizes: ('default', 'lg', 'sm', 'xs', '2xs');
+$body-sizes: ('default', 'default-emphasized', 'lg', 'lg-emphasized', 'sm', 'sm-emphasized', 'xs', 'xs-emphasized', '2xs', '2xs-emphasized');
```
Nit: "emphasized" is a mouthful. Might be too late to shorten to something like `emph` across the repo, but thought I'd throw it out there.

If it diverges from Figma parity, it's probably not worth it.

jordanjones243 replied:
I don't disagree, but definitely a design decision. Let me know what you think @forrest-for-real.

forrest-for-real replied:
> I don't disagree, but definitely a design decision. Let me know what you think @forrest-for-real.

I think we should keep it as that's how it is in Figma, and if I can figure out how to export type directly from Figma, it will be useful. It's a mouthful but descriptive.

jordanjones243 committed:
refactor: simplify fluid type and accent mixins by removing theme-name parameter

chrisfalaska commented on src/type/mixins/_theme-generator.scss:
```diff
@@ -14,9 +14,12 @@
   );
   
+  // Derive letter-spacing behavior from accent config data rather than theme name
+  $accent-uses-letter-spacing: type-property-check($accent-config, '2xl', 'letter-spacing') != null;
```
This is defining the pattern for all sizes based on just `2xl`.

I would suggest this file look like this:

```
@import "../partials/body";
@import "../partials/display";
@import "../partials/heading";
@import "../partials/accent";
@import "../mixins/type-generator";

// Theme generator
@mixin generate-theme(
  $theme-name,
  $body-config,
  $display-config,
  $heading-config,
  $accent-config,
  $accent-uses-letter-spacing: true
) {
  // Map of all type configs for the theme
  $theme-configs: (
    'body': $body-config,
    'display': $display-config,
    'heading': $heading-config,
    'accent': $accent-config
  );

  // Generate CSS variables for the theme
  @include generate-theme-type-css-vars($theme-configs);

  // Generate theme type classes
  @include generate-theme-type-classes($use-letter-spacing: $accent-uses-letter-spacing);
}
```

And then in `src/type/themes/hawaiian` it would look like this:

```
// Generate Hawaiian theme
@include generate-theme(
  'hawaiian',
  $hawaiian-type-body-config,
  $hawaiian-type-display-config,
  $hawaiian-type-heading-config,
  $hawaiian-type-accent-config,
  $accent-uses-letter-spacing: false
);
```

Where we just pass in `$accent-uses-letter-spacing: false`

chrisfalaska commented on src/type/mixins/_fluid-type.scss:
```diff
@@ -9,10 +9,13 @@
+
+    @if type-property-check($theme-config, $size, 'letter-spacing') != null {
+      --#{$css-ns}-#{$type}-#{$size}-letter-spacing: #{type-property-check($theme-config, $size, 'letter-spacing')};
+    }
```
I don' think this is necessary with `$use-letter-spacing` at the theme level. I would maybe remove this `@if` and this on line `12`:

```
--#{$css-ns}-#{$type}-#{$size}-letter-spacing: #{type-property-check($theme-config, $size, 'letter-spacing')};
```

chrisfalaska committed:
refactor: replace inferred letter-spacing flag with explicit theme parameter

Surface accent letter-spacing intent at the theme call site instead of
inferring it from the accent-config 2xl entry. The inference was brittle:
a future theme defining letter-spacing on some sizes but not 2xl would
have been miscategorized for the entire accent group.

chrisfalaska committed:
refactor: emit letter-spacing CSS var unconditionally in fluid-type

The per-size guard on var emission and the theme-level $use-letter-spacing
flag on class emission were two independent gates that could disagree. A
mixed config could produce a class referencing an undefined var, causing
letter-spacing to resolve to unset rather than inherit. Make the theme
flag the single source of truth.

chrisfalaska committed:
refactor: rename generate-theme-type-classes flag to clarify accent scope

The mixin name suggests it covers all four type categories, but the
$use-letter-spacing parameter only ever flowed to generate-accent-classes.
Rename to $accent-use-letter-spacing so the scope is visible at the call
site without stepping into the mixin body.

jordanjones243 commented:
5/1 Diff Checks:

Hawaiian - https://www.diffchecker.com/QtxiW6Mb/

Alaska - https://www.diffchecker.com/rNhRQigc/

Atmos - https://www.diffchecker.com/fMqQU8pp/

chrisfalaska commented on src/type/partials/_body.scss:
```diff
@@ -3,16 +3,16 @@
 @import '../../utilityVariables/cssNamespace';
 
-$body-sizes: ('default', 'lg', 'sm', 'xs', '2xs');
+$body-sizes: ('default', 'default-emphasized', 'lg', 'lg-emphasized', 'sm', 'sm-emphasized', 'xs', 'xs-emphasized', '2xs', '2xs-emphasized');
```
insted of adding duplicate sizes, we could update this to something like this which scales from a single source of truth:

```
$body-base-sizes: ('default', 'lg', 'sm', 'xs', '2xs');

$body-sizes: ();
@each $size in $body-base-sizes {
  $body-sizes: append($body-sizes, $size);
  $body-sizes: append($body-sizes, '#{$size}-emphasized');
}
```

jordanjones243 committed:
refactor: optimize function to remove duplication

chrisfalaska reviewed (approved):
Looks great! 🚀
